server/src/main/java/game/GameConfig.java

```diff
@@ -35,6 +38,14 @@ public ChunkInfo(double size, int n) {
         }
     }
 
+    public record BotPopulationDistributionTarget(double stupidBotRatio, double kamikazeBotRatio,
+                                                  double scaredBotRatio) {
+        public BotPopulationDistributionTarget {
+            assert (stupidBotRatio + kamikazeBotRatio + scaredBotRatio == 1.0);
```

Comment:
Careful with exact floating point math. Better check for `Math.abs(x1 + x2 + x3 - 1.0) < 1e-8`